iceoryx2-ffi/cxx/include/iox2/sample.hpp

```diff
@@ -49,8 +54,13 @@ class Sample {
     auto operator->() const -> const Payload*;
 
     /// Returns a reference to the payload of the [`Sample`]
+    template <typename T = Payload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, void>>
     auto payload() const -> const Payload&;
 
+    /// Returns a slice to navigate the payload of the [`Sample`]
+    template <typename T = Payload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, void>>
+    auto payload_slice() const -> iox::ImmutableSlice<const ValueType>;
```

Comment:
I think this can now be a `iox::ImmutableSlice<ValueType>`

Reply:
Fixed